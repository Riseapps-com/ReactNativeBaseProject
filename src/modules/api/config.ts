const AXIOS_REQUEST_TIMEOUT_MSEC = 16000;

export default AXIOS_REQUEST_TIMEOUT_MSEC;
