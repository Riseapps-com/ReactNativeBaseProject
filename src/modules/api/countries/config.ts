export const ALL = 'all';
export const BY_REGION = 'region';
export const BY_CODE = 'alpha';
