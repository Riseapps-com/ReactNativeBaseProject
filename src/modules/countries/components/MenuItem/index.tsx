import React, { useCallback } from 'react';
import { View } from 'react-native';

import { LocalImage, Text } from '~modules/ui';
import { useTheme } from '~theme';

import Card from '../Card';

import themedStyles from './styles';

import type { Images } from '~assets';
import type { StyleProp, ViewStyle } from 'react-native';

export type MenuItemProps = {
  style?: StyleProp<ViewStyle>;
  image: keyof Images;
  title: string;
  index: number;
  onItemPress?: (index: number) => void;
  testID?: string;
};

const MenuItem: React.FC<MenuItemProps> = props => {
  const [styles] = useTheme(themedStyles);
  const { onItemPress } = props;

  const handleItemPress = useCallback(() => onItemPress && onItemPress(props.index), [onItemPress, props.index]);

  return (
    <Card
      testID={props.testID}
      style={[styles.container, props.style]}
      contentStyle={styles.contentContainer}
      onPress={handleItemPress}
      activeOpacity={0.7}
    >
      <View style={styles.imageContainer}>
        <LocalImage style={styles.image} source={props.image} />
      </View>

      <View style={styles.titleContainer}>
        <Text numberOfLines={2} fontStyle="bold" style={styles.title}>
          {props.title}
        </Text>
      </View>
    </Card>
  );
};

export default MenuItem;
