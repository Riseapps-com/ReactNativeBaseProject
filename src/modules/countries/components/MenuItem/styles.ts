import { Platform } from 'react-native';

import { ELEVATION, SHADOW_HEIGHT, SHADOW_OPACITY, SHADOW_RADIUS } from '~modules/ui';
import { sizes, styleSheetFactory } from '~theme';

const MENU_IMAGE_WIDTH = 48;

export default styleSheetFactory(theme => ({
  container: {
    ...Platform.select({
      ios: {
        shadowColor: theme.backdrop,
        shadowOffset: {
          width: 0,
          height: SHADOW_HEIGHT,
        },
        shadowOpacity: SHADOW_OPACITY,
        shadowRadius: SHADOW_RADIUS,
      },
    }),
    marginHorizontal: sizes.SPACING_S,
    marginVertical: sizes.SPACING_S,
    padding: sizes.SPACING_S,
    flex: 1,
  },

  contentContainer: {
    ...Platform.select({ android: { elevation: ELEVATION } }),
    overflow: 'hidden',
    borderRadius: sizes.SPACING_S,
    flexDirection: 'row',
    backgroundColor: theme.primary,
    flex: 1,
  },

  imageContainer: {
    width: '30%',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: sizes.SPACING_S,
  },

  image: {
    width: MENU_IMAGE_WIDTH,
    height: MENU_IMAGE_WIDTH,
  },

  titleContainer: {
    flex: 1,
    justifyContent: 'center',
    paddingStart: sizes.SPACING_S,
    paddingEnd: sizes.SPACING_M,
  },

  title: {
    color: theme.secondaryText,
  },
}));
