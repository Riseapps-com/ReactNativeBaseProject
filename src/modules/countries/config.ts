import { LocalRegion } from './types';

export const STORYBOOK_SCREEN_NAME = 'STORYBOOK_SCREEN';
export const MENU_SCREEN_NAME = 'MENU_SCREEN';
export const COUNTRIES_SCREEN_NAME = 'COUNTRIES_SCREEN';
export const SELECT_REGION_SCREEN_NAME = 'SELECT_REGION_SCREEN';
export const COUNTRY_DETAILS_SCREEN_NAME = 'COUNTRY_DETAILS_SCREEN';

export const regions: LocalRegion[] = ['africa', 'americas', 'asia', 'europe', 'oceania'];
