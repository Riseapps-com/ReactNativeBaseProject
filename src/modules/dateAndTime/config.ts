export const LOG_DATE_TIME_FORMAT = 'YYYY-MM-DD HH:mm:ss.SSSZ';
