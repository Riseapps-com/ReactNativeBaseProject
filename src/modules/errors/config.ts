import type { ErrorCode } from './types';

export const RETRYABLE_ERROR_CODES: ErrorCode[] = ['UnknownException', 'ServerErrorException'];
