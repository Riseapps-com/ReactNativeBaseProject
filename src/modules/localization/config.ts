export const LOCALIZATION_FALLBACK_LANGUAGE = 'en';
