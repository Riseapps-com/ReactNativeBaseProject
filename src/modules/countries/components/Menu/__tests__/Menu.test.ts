import { useNavigation } from '@react-navigation/native';
import { fireEvent } from '@testing-library/react-native';

import { testIDs } from '~config';
import { I18n } from '~modules/localization';
import { mocked, renderComponent } from '~modules/tests';

import { COUNTRIES_SCREEN_NAME, SELECT_REGION_SCREEN_NAME } from '../../../config';
import MenuItem from '../index';

const mockedUseNavigation = mocked(useNavigation);

const renderMenu = () => renderComponent(MenuItem);

describe('countries', () => {
  describe('<Menu />', () => {
    it('renders <Menu />', () => {
      const menu = renderMenu();

      expect(menu.getByTestId(testIDs.menu.allCountries)).toBeTruthy();
      expect(menu.getByTestId(testIDs.menu.countriesByRegion)).toBeTruthy();
      expect(menu).toMatchSnapshot();
    });

    it('navigates to countries list', () => {
      const menu = renderMenu();

      fireEvent.press(menu.getByTestId(testIDs.menu.allCountries));
      jest.runAllTimers();

      expect(mockedUseNavigation().navigate).toBeCalledTimes(1);
      expect(mockedUseNavigation().navigate).toBeCalledWith(COUNTRIES_SCREEN_NAME, {
        title: I18n.t('allCountries'),
      });
    });

    it('navigates to select region menu', () => {
      const menu = renderMenu();

      fireEvent.press(menu.getByTestId(testIDs.menu.countriesByRegion));
      jest.runAllTimers();

      expect(mockedUseNavigation().navigate).toBeCalledTimes(1);
      expect(mockedUseNavigation().navigate).toBeCalledWith(SELECT_REGION_SCREEN_NAME);
    });
  });
});
