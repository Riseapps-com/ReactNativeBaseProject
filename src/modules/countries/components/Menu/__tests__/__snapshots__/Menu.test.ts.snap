// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`countries <Menu /> renders <Menu /> 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 16,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="menu_allCountries"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/flag.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          allCountries
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 8,
        "marginVertical": 16,
        "opacity": 1,
        "padding": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
    testID="menu_countriesByRegion"
  >
    <View
      style={
        Object {
          "backgroundColor": "#0036A7",
          "borderRadius": 8,
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingHorizontal": 8,
            "width": "30%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/region.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingEnd": 16,
            "paddingStart": 8,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#4A4E4D",
              },
              Object {
                "fontFamily": "Quicksand-Bold",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          countriesByRegion
        </Text>
      </View>
    </View>
  </View>
</View>
`;
